src/SFA.DAS.AssessorService.Web/Controllers/CertificateOptionController.cs

```diff
@@ -135,5 +134,62 @@ private string GetModelValues<T>(T viewModel)
 
             return properties.Aggregate("", (current, prop) => current + $"{prop.Name}: {prop.GetValue(viewModel)}, ");
         }
+
+        [HttpGet("back", Name = "CertificateOptionBack")]
+        public IActionResult Back()
+        {
+            var username = ContextAccessor.HttpContext.User.FindFirst("http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn")?.Value;
+            var sessionString = SessionService.Get(nameof(CertificateSession));
+            if (sessionString == null)
+            {
+                Logger.LogInformation($"Session for CertificateOptionViewModel requested by {username} has been lost. Redirecting to Search Index");
+                return RedirectToAction("Index", "Search");
+            }
+            var certSession = JsonConvert.DeserializeObject<CertificateSession>(sessionString);
+
+            var hasVersions = certSession.Versions != null && certSession.Versions.Any();
+
+            if (hasVersions)
+            {
+                //redirectToCheck returns you to the check certificate page
+                //redirectToVersion is where a user has gone to the check page, selected change version
+                //has changed version, then has to pick a new option as well if they exist
+                //as a result redirecttocheck is passed so on selection of option the user goes back to the check page
+                //however if a user selects back, they must go back to the version select page.
+                //If a user on the check page changes just the option, then that should just go 
+                //back to the check page.
+                //redirectfromversion is always set when coming from version page to allow for this
+                //however it's removed beyond that page, therefore if coming back from declaration
+                //the final version count check allows that fallback.
+                var redirectToCheck = SessionService.Exists("redirecttocheck") && bool.Parse(SessionService.Get("redirecttocheck"));
```

Comment:
this seems like it could be encapsulated in a `SessionService.TryGet<bool>("redirecttocheck")` ?

Reply:
I've implemented a version of TryGet to simplify this code.